src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -770,27 +770,22 @@ public static void Reverse(ref char buf, nuint length)
                     tempLast = Avx2.Permute2x128(tempLast, tempLast, 0b00_01);
 
                     // Store the reversed vectors
-                    tempLast.StoreUnsafe(ref bufByte, firstOffset);
-                    tempFirst.StoreUnsafe(ref bufByte, lastOffset);
-                }
-                bufByte = ref Unsafe.Add(ref bufByte, numIters * numElements);
-                length -= numIters * (nuint)Vector256<short>.Count * 2;
-                // Store any remaining values one-by-one
-                buf = ref Unsafe.As<byte, char>(ref bufByte);
+                    tempLast.StoreUnsafe(ref Unsafe.As<char, byte>(ref first));
+                    tempFirst.StoreUnsafe(ref Unsafe.As<char, byte>(ref last));
+
+                    first = ref Unsafe.Add(ref first, Vector256<ushort>.Count);
+                    last = ref Unsafe.Subtract(ref last, Vector256<ushort>.Count);
+                } while (Unsafe.IsAddressLessThan(ref first, ref last));
+
+                remainder = Unsafe.ByteOffset(ref first, ref Unsafe.Add(ref last, Vector256<ushort>.Count)) / sizeof(ushort);
```

Comment:
It would be better to keep `remainder` as `nuint` so you're not doing a signed division here.  (It still gets implemented as a shift but with extra steps to handle the sign).

Reply:
Thanks!